zoneinfo/src/utils.rs

```diff
@@ -0,0 +1,63 @@
+//! Utiltiy functions
+
+pub const SECONDS_PER_DAY: i64 = 24 * 60 * 60;
+
+pub(crate) fn mathematical_days_in_year(y: i32) -> i32 {
+    if y % 4 != 0 {
+        365
+    } else if y % 4 == 0 && y % 100 != 0 {
+        366
+    } else if y % 100 == 0 && y % 400 != 0 {
+        365
+    } else {
+        // Assert that y is divisble by 400 to ensure we are returning the correct result.
+        assert_eq!(y % 400, 0);
+        366
+    }
+}
+
+pub(crate) fn in_leap_year(year: i32) -> i32 {
+    mathematical_days_in_year(year) - 365
+}
```

Comment:
suggestion: Rename this or have it return `bool`. The way this is used is a little surprising. Maybe `num_leap_days()`.

Reply:
Yeah, this was something pulled over from the Temporal spec :laughing: